### PR TITLE
Support feature normalization ops on edge features.

### PR DESCRIPTION
**Pull request by classicsong**
*Issue #, if available:*
#259 

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/gconstruct/construct_graph.py [hunk 1]
```diff
@@ -34,20 +34,18 @@
...
-                        GlobalProcessFeatTransform)
+from .transform import do_multiprocess_transform
 from .id_map import NoopMap, IdMap, map_node_ids
 from .utils import (multiprocessing_data_read,
                     update_two_phase_feat_ops, ExtMemArrayMerger,
                     partition_graph)
 
 def prepare_node_data(in_file, feat_ops, read_file):
-    """ Parse node data.
+    """ Prepare node data for feature transform
```
Comment (zheng-da) [inline python/graphstorm/gconstruct/construct_graph.py:44]: transform => transformation

File: python/graphstorm/gconstruct/utils.py [hunk 2]
```diff
@@ -378,8 +378,15 @@ def __call__(self, arrs, name):
         if len(arrs) > 1:
             return _merge_arrs(arrs, tensor_path)
         else:
+            # To get the output dtype by accessing the
+            # first element of the arrays (numpy array or HDFArray)
+            # Note: We use arrs[0][0] instead of arrs[0] because
+            #       arrs[0][0] is a transformed data with out_dtype
+            #       while arrs[0] can be a HDFArray and has not
+            #       been cast to out_dtype.
+            dtype = arrs[0][0].dtype
```
Comment (zheng-da) [inline python/graphstorm/gconstruct/utils.py:387]: how many places do you need to change it like this?

Comment (zheng-da) [reply python/graphstorm/gconstruct/utils.py:387]: maybe write a wrapper function?

File: python/graphstorm/gconstruct/construct_graph.py [hunk 3]
```diff
@@ -101,6 +99,32 @@ def parse_node_data(in_file, feat_ops, label_ops, node_id_col, read_file):
     node_ids = data[node_id_col] if node_id_col in data else None
     return (node_ids, feat_data)
 
+def prepare_edge_data(in_file, feat_ops, read_file):
+    """ Parse node data.
```
Comment (zheng-da) [inline python/graphstorm/gconstruct/construct_graph.py:103]: we need to change the docstring. how is this different from `parse_edge_data`?

**Review by zheng-da — APPROVED** [review]

File: python/graphstorm/gconstruct/utils.py (hunk 2 above)
Comment (classicsong) [reply python/graphstorm/gconstruct/utils.py:387]: only 2

File: python/graphstorm/gconstruct/construct_graph.py (hunk 3 above)
Comment (classicsong) [reply python/graphstorm/gconstruct/construct_graph.py]: Updated

File: python/graphstorm/gconstruct/utils.py (hunk 2 above)
Comment (classicsong) [reply python/graphstorm/gconstruct/utils.py]: I agree, we need a wrapper function. This part of the code is tricky.

**Pull request merged**